### ISSUE TITLE
Dataset creation in the paper

### ISSUE THREAD

**Issue by hamedbehzadi** [frankwang345/class-subnetwork#1]
Dear Dr. Wang

I hope this message finds you well.
My name is Hamed Behzadi and I am a Ph.D. student at the University of Antwerp.

For one of my projects, I found your recently published paper with the title "Interpret Neural Networks by Extracting Critical Subnetworks" very interesting and inspiring. The idea is so novel and properly understandable and can be added to a very wide range of CNNs. I have tried to implement it. However, I have faced an issue. When I want to extract binary subnetworks, I need to convert multi-classes dataset to binary dataset. As a result, for training the gates of the subnetowrk, I face an imbalanced dataset that effect on the subnetwork accuracy performance.

I was wondering if you could please guide me in this regard that how did you define your binary dataset. Have you considered all of the samples from other classes as negative class or just a subset of images from other classes equal to the number of images in the target class(positive class)?
Also, if there is still any implementation of this work, I really would appreciate it if you could share it with me.

Thank you for your attention and time given to my request.

Kind regards,
Hamed
